### ISSUE TITLE
Interpolation and creating texture coordinate buffers

### ISSUE THREAD

**Issue by LegendWasTaken** [embree/embree#325]
I'm attempting to get texture coordinates working on my path tracer. And have found a lack of documentation on user vertex attribute data. I've found out-dated information for Embree2. But nothing for Embree 3, how would one go about adding per-vertex data to be interpolated later?

I should note: The texture coordinates I'm using have their own index buffer. And there's no documentation I've found that shows you how to create a GeometryAttribute buffer, and attach an IndexBuffer to it. 

This is my current code - And the `rtcSetNewGeometryBuffer` fails when I try to create the `BUFFER_TYPE_INDEX`

   ` 
auto *tex_index_buffer = static_cast<uint32_t *>(rtcSetNewGeometryBuffer(
instance.geometry,
RTC_BUFFER_TYPE_INDEX,
2,
RTC_FORMAT_UINT3,
sizeof(uint32_t) * 3,
tex_indices.size() / 3));`

Tldr; I want to create a seperate a buffer for `RTC_BUFFER_TYPE_VERTEX_ATTRIBUTE` in a seperate slot, so it can have it's own index buffer. But I'm not able to find any documentation on it

**Comment by cbenthin** [comment]
Typically I would refer you to the "interpolation" tutorial which uses rtcInterpolate to interpolate per vertex data. The problem in your case is that the separate index buffer for texture coordinates. I would suggest to create a "shadow" Embree object using this index buffer and then interpolate the texture coordinates using rtcInterpolate.

**Comment by LegendWasTaken** [comment]
Create a "shadow" embree object, what does this mean? Create a different geometry with the indices and then use the primID on that geometry?

**Comment by svenwoop** [comment]
You likely use triangle meshes? In this case I would recommend not using rtcInterpolate but to interpolate the vertex data manually in the app (this is faster). the rtcInterpolate function is best just used for Subdivision Meshes.
Use this formula for interpolation (1-u-w)*t0+ u*t1 + v*t2, where t0,t1,t2 are the texture coordinates at vertices 0,1,2
